### PR TITLE
[Php80] Handle nested annotation not end with () on AnnotationToAttributeRector 

### LINKED ISSUE #7558: AnnotationToAttributeRector converts nested annotations without `()` to string
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | dev-main@b9c04e      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/5e212222-e03c-416b-85bf-633aa27b4f9b
Not the best example, but I used multiple `@Assert\NotBlank` annotations to make it clear which cases are converted correctly 🙂 

Nested annotations, without trailing `()`, are converted to strings by the AnnotationToAttributeRector:

```
// From \Rector\Symfony\Set\SymfonySetList::SYMFONY_54:
$rectorConfig->ruleWithConfiguration(AnnotationToAttributeRector::class, [
    new AnnotationToAttribute('Symfony\\Component\\Validator\\Constraints\\Sequentially')
]);
```

```php
<?php
use Symfony\Component\Validator\Constraints as Assert;
use App\Validator\Constraints as UserAssert;

class User
{
    /**
     * @Assert\NotBlank
     * @Assert\Sequentially({
     *     @Assert\NotBlank,
     *     @Assert\NotBlank()
     * })
     */
     private string $email;
}
```

### Responsible rules

* `AnnotationToAttributeRector`

## Expected Behavior

I expect that Rectors can converts annotations with and without trailing `()` to attributes:

```diff
-     #[Assert\Sequentially(['@Assert\NotBlank', new Assert\NotBlank()])]
+     #[Assert\Sequentially([@Assert\NotBlank(), new Assert\NotBlank()])]
````

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/7558

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
         "symfony/dependency-injection": "^6.1.3",
         "symfony/finder": "^6.1.3",
         "symfony/string": "^6.1.4",
-        "symplify/autowire-array-parameter": "^11.1.10",
+        "symplify/autowire-array-parameter": "dev-main as 11.1.15",
```
Comment (samsonasik) [inline composer.json]: @TomasVotruba this is temporary needed while wait for next symplify/autowire-array-parameter release, ref https://github.com/symplify/symplify/pull/4458

**Comment by samsonasik** [comment]
CI error seems unrelated, even on existing `symplify/autowire-array-parameter: 11.1.15`

```bash
➜  rector-src git:(close-7558) ✗ bin/rector process packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser/PlainValueParser.php --dry-run
PHP Fatal error:  Uncaught _PHPStan_71ced81c9\Nette\DI\ServiceCreationException: Multiple services of type PHPStan\Rules\PHPUnit\CoversHelper found: 0293, 0305 in phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Autowiring.php:53
Stack trace:
#0 phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/ContainerBuilder.php(170): _PHPStan_71ced81c9\Nette\DI\Autowiring->getByType('PHPStan\\Rules\\P...', true)
#1 phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolver.php(312): _PHPStan_71ced81c9\Nette\DI\ContainerBuilder->getByType('PHPStan\\Rules\\P...', true)
#2 phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolver.php(137): _PHPStan_71ced81c9\Nette\DI\Resolver->getByType('PHPStan\\Rules\\P...')
#3 phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolver.php(443): _PHPStan_71ced81c9\Nette\DI\Resolver->_PHPStan_71ced81c9\Nette\DI\{closure}('PHPStan\\Rules\\P...', true)
#4 phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolve
```

File: composer.json (hunk 1 above)
Comment (TomasVotruba) [reply composer.json]: Release is on the way :+1:  ETA 5 mins

Comment (samsonasik) [reply composer.json]: Thank you, rolled back https://github.com/rectorphp/rector-src/pull/3017/commits/6a4947446130208f54b9b688f10bc0a6a9527ad3

**Comment by samsonasik** [comment]
@TomasVotruba CI seems still error https://github.com/rectorphp/rector-src/actions/runs/3336708101/jobs/5522181188#step:6:9

```bash
Run bin/rector process src tests rules-tests packages packages-tests --ansi
PHP Fatal error:  Uncaught _PHPStan_71ced81c9\Nette\DI\ServiceCreationException: Multiple services of type PHPStan\Rules\PHPUnit\CoversHelper found: 0293, 0305 in phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Autowiring.php:53
Stack trace:
#0 phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/ContainerBuilder.php(170): _PHPStan_71ced81c9\Nette\DI\Autowiring->getByType()
#1 phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolver.php(312): _PHPStan_71ced81c9\Nette\DI\ContainerBuilder->getByType()
#2 phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolver.php(137): _PHPStan_71ced81c9\Nette\DI\Resolver->getByType()
#3 phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolver.php(443): _PHPStan_71ced81c9\Nette\DI\Resolver->_PHPStan_71ced81c9\Nette\DI\{closure}()
#4 phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolver.php(410): _PHPStan_71ced81c9\Nette\DI\Resolver::autowireArgument()
#5 phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Resolver.php(170): _PHPStan_71ced81c9\Nette\DI\Resolver::autowireArguments()
#6 phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Definitions/ServiceDefinition.php(158): _PHPStan_71ced81c9\Nette\DI\Resolver->completeStatement()
```

The `PHPStan\Rules\PHPUnit\CoversHelper` seems defined once, while it show error defined multiple

**Comment by samsonasik** [comment]
@TomasVotruba pin `phpstan/phpstan-phpunit` to `1.1.3` seems solve it https://github.com/rectorphp/rector-src/pull/3017/commits/e401131d2abde82a381ef9be793e25c0ad5eb074

**Comment by samsonasik** [comment]
The error is now only on tests, on `ParamAnnotationIncorrectNullableRector`, which caused by changed code:

```bash
There were 3 failures:

1) Rector\Tests\TypeDeclaration\Rector\ClassMethod\ParamAnnotationIncorrectNullableRector\ParamAnnotationIncorrectNullableRectorTest::test with data set #20 ('/home/runner/work/rector-src/...hp.inc')
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
      *     @Assert\NotBlank,
      *     @AppAssert\Country,
      * })
-     * @param \DateTime[]|null $dateTimes
+     * @param \DateTime[] $dateTimes
```

**Comment by samsonasik** [comment]
The patch seems need to be in ArrayItemNodeAnnotationToAttributeMapper https://github.com/rectorphp/rector-src/pull/3017/commits/1c32de884fc4feed141a056e05db58366567a33f

**Comment by samsonasik** [comment]
Finally 🎉 🎉 🎉 All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Thank you :clap:

**Pull request merged**